docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,258 @@
+code, and some of these use cases would require maintaining a vendor
+copy of the code. Adding these features requires rather extensive
+(internal/implementation) changes to the `node` and `rpc` packages,
+and this ADR describes a model for changing the way that tendermint
+nodes initialize, in service of providing this kind of functionality.
+
+We consider node initialization, because the current implemention
+provides strong connections between all components, as well as between
+the components of the node and the RPC layer, and being able to think
+about the interactions of these components will help enable these
+features and help define the requirements of the node package.
+
+## Alternative Approaches
+
+### Do Nothing
+
+The current implementation is functional and sufficient for the vast
+majority of use cases (e.g. all users of the Cosmos-SDK as well as
+anyone who runs tendermint and the ABCI application in separate
+processes.) In the current implementation, and even previous versions,
+modifying node initialization or injecting custom components required
+copying most of the `node` package, which amounts to requiring users
+to maintain a vendored copy of tendermint for these kinds of use
+cases.
+
+While this is (likely) not tenable in the long term, as users do want
+more modularity, and the current service implementation is brittle and
+difficult to maintain, in the short term it may be possible to delay
+implementation somewhat. Eventually, however, we will need to make the
+`node` package easier to maintain and reason about.
+
+### Generic Service Plugability
+
+We can imagine a system design that exports interfaces (in the Golang
+sense) for all components of the system, to permit runtime dependency
+injection of all components in the system so that users can compose
+tendermint nodes of arbitrary user supplied components.
+
+This is an interesting proposal, and customization would provide
+benefits, this is a huge undertaking particularly with regards to API
+design work, that we do not have scope for at the moment. While this
+is not a goal at the moment, eventually providing support for some
+kinds of plugability may be useful, so the current solution does not
+explicitly forclose the possibility of this alternative.
+
+### Abstract Dependency Based Startup and Shutdown
+
+The proposal on in this document simplifies and makes tendermint node
+initialization more abstract, but the system lacks a number of
+features which daemon/service initialization might provide, such as a
+dependency based system that allows the authors of services to control
+the initialization and shutdown order of components using an
+dependencies to control the ordering.
+
+Such a system would work, by allowing services to declare
+initialization order dependencies to other reactors (by ID, perhaps)
+so that the node could decide the initialization based on the
+dependencys declared by components rather than requiring the node to
```

Comment:
```suggestion
dependencies declared by components rather than requiring the node to
```

I'm not sure I understand what  a "component" is in this context—do you mean "declared via application-level configuration settings" perhaps?